### PR TITLE
PLANNER-366: Workbench: Construction Heuristics solver configuration

### PR DESCRIPTION
**Pull request by mcimbora**
This PR provides base for adding Solver phases in Solver configuration section. Only Construction heuristic phases are supported ATM (will be extended with Local search, etc). There might be some more UI changes as I receive feedback from UXD.

### CODE REVIEW

**Comment by mcimbora** [comment]
@ge0ffrey Regarding `ConstructionHeuristicTypeModel` enum, I do believe we can reuse OptaPlanner enum (after making it JS-friendly). I'll have a look at this.

File: optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/ToSolverConfig.java [hunk 1]
```diff
@@ -116,4 +122,23 @@ private TerminationConfig create( final TerminationConfigModel termination ) {
             return terminationConfig;
         }
     }
+
+    private List<PhaseConfig> create( final List<PhaseConfigModel> phaseConfigList ) {
+        if ( phaseConfigList == null ) {
+            return null;
```
Comment (csadilek) [inline optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/ToSolverConfig.java]: ToSolverConfigModel.create returns an empty list, this returns null. Wouldn't it be better to return an empty list as well?

File: optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/test/java/org/optaplanner/workbench/screens/solver/backend/server/ToSolverConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import static org.junit.Assert.assertEquals;
+
+public class ToSolverConfigTest {
+
+    @Test
+    public void get() {
+        ToSolverConfig toSolverConfigModel = new ToSolverConfig( getSolverConfigModel() );
+        SolverConfig solverConfig = toSolverConfigModel.get();
+
+        assertEquals( Long.valueOf( 1 ), solverConfig.getTerminationConfig().getMillisecondsSpentLimit() );
```
Comment (csadilek) [inline optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/test/java/org/optaplanner/workbench/screens/solver/backend/server/ToSolverConfigTest.java]: IMO, It'd be easier to read after assigning solverConfig.getTerminationConfig() to a local variable.

**Comment by csadilek** [comment]
+1 (just some minor comments, see inline)

**Comment by mcimbora** [comment]
@csadilek Thanks, fixed.

**Pull request merged**